### ISSUE TITLE
Improve search tests

### ISSUE THREAD

**Issue by curquiza** [meilisearch/meilisearch-dotnet#71]
This issue is to improve the tests relative to the `Search` method. It means improving [this file](https://github.com/meilisearch/meilisearch-dotnet/blob/master/tests/Meilisearch.Tests/SearchTests.cs) by adding tests:
- do tests about all the [search parameters](https://docs.meilisearch.com/guides/advanced_guides/search_parameters.html). For example, there are no tests about the crop.
<s>- add better `Filters` tests with `AND` and `OR` (see the end of this [page of docs](https://docs.meilisearch.com/guides/advanced_guides/filtering.html))</s>

[This file](https://github.com/meilisearch/meilisearch-dotnet/blob/master/tests/Meilisearch.Tests/IndexFixture.cs) might need changes to fit those new tests.

**Comment by anthkernan-dev** [comment]
@brunoocasali - Is this still a valid issue? If it is, I've looked at it today but the attributes that i'm specifying in the search query are always returned in full in the _formatted object.

**Comment by brunoocasali** [comment]
Looking into this diff (the day which Clem created this issue and current main version) https://github.com/meilisearch/meilisearch-dotnet/compare/b243cf243336901a2218a69be827019cdb8c4ab8...main#

I could check that we added a lot of tests: 

`CustomSearchWithSort`, `CustomSearchWithPhraseSearch`, `CustomSearchWithMultipleFilter`, `CustomSearchWithNumberFilter`, `CustomSearchWithFilterArray`, `CustomSearchWithFilterWithSpaces`, `CustomSearchWithFilter`.

But still, we have some of them missing, if you look into other SDK's (ruby for eg. https://github.com/meilisearch/meilisearch-ruby/blob/main/spec/meilisearch/index/search/attributes_to_crop_spec.rb we have a spec for `attributesToCrop` which is missing from dotnet repo, `matches` also and so on...). 

In summary, we want to have all the options `src/Meilisearch/SearchQuery.cs` covered.

**Comment by brunoocasali** [comment]
If you want to, you could try to cover some of these cases, we appreciate the help 😸 

Let me know if you need anything @AK0785 !

**Comment by anthkernan-dev** [comment]
@brunoocasali  - I can't seem to return a matches object even though I've set the matches parameter to true and it is being passed into the SearchQuery with a string that is contained in either the Title or Genre?

**Comment by brunoocasali** [comment]
Sorry @AK0785 I spoke to @curquiza but we weren't able to understand what you meant in your last message, can you give more details or even an example about what you mean?
